src/box/iproto.cc

```diff
@@ -1857,7 +2274,88 @@ net_send_msg(struct cmsg *m)
 {
 	struct iproto_msg *msg = (struct iproto_msg *) m;
 	struct iproto_connection *con = msg->connection;
+	uint32_t stream_id = msg->header.stream_id;
+	mh_int_t pos;
+	struct stream *stream;
+
+	rlist_del_entry(msg, in_pending_list);
+	/*
+	 * If connection was closed, we stop processed remaining
+	 * stream messages, they will be deleted with stream in
+	 * `net_finish_disconnect`
+	 */
+	if (stream_id == 0 || con->state == IPROTO_CONNECTION_CLOSED)
+		goto send_msg;
+
+	pos = mh_streams_find(con->streams, stream_id, 0);
+	assert(pos != mh_end(con->streams));
+	stream = *mh_streams_node(con->streams, pos);
+
+	if (rlist_empty(&stream->pending_list)) {
+		/*
+		 * If no more messages for the current stream
+		 * and no transaction started, then delete it.
+		 */
+		if (stream->txn == NULL) {
+			mh_streams_remove(con->streams, (const struct stream **)
+					  &stream, 0);
+			iproto_stream_delete(stream);
+		}
+	} else {
+		/*
+		 * If there are new messages for this stream
+		 * then schedule their processing.
+		 */
+		struct iproto_msg * next =
```

Comment:
Star must be immediately before variable name.

Reply:
Fixed